daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DaffodilParseOutputStreamContentHandler.scala

```diff
@@ -116,9 +247,16 @@ class DaffodilParseOutputStreamContentHandler(out: OutputStream, pretty: Boolean
       }
     }
     writer.write("</")
-    writer.write(qName)
+    outputTagName(uri, localName, qName)
     writer.write(">")
     outputNewlineStack.pop()
+
+    if (!globalPrefixMappingContextStack.isEmpty) globalPrefixMappingContextStack.pop
+    if (globalPrefixMappingContextStack.isEmpty) {
+      globalPrefixMapping = null
+    } else {
+      globalPrefixMapping = globalPrefixMappingContextStack.top
+    }
```

Comment:
I think we searched for one and actually couldn't find many that wrote out to an OutputStream. And those that did pulled in tons of dependencies or didn't work the way we expected. Might be worth a second look. I think we also recently added SaxonHE as a new dependency, so that might be worth checking as well.

Reply:
I looked into SaxonHE, and I couldn't find an outputStream contentHandler.